lldb/source/Plugins/LanguageRuntime/Swift/SwiftTask.cpp

```diff
@@ -0,0 +1,62 @@
+#include "SwiftTask.h"
+#include "SwiftLanguageRuntime.h"
+#include "lldb/Target/Process.h"
+
+using namespace llvm;
+using namespace lldb;
+
+namespace lldb_private {
+
+ThreadTask::ThreadTask(tid_t tid, addr_t async_ctx, ExecutionContext &exe_ctx)
+    : Thread(exe_ctx.GetProcessRef(), tid, true),
+      m_reg_info_sp(exe_ctx.GetFrameSP()->GetRegisterContext()) {
+  m_async_ctx = async_ctx;
+  uint32_t ptr_size =
+      exe_ctx.GetTargetRef().GetArchitecture().GetAddressByteSize();
+  // A simplified description of AsyncContext. See swift/Task/ABI.h
+  // struct AsyncContext {
+  //   AsyncContext *Parent;                    // offset 0
+  //   TaskContinuationFunction *ResumeParent;  // offset 8
+  // };
+  uint32_t resume_offset = ptr_size; // offsetof(AsyncContext, ResumeParent)
+  uint32_t resume_ptr = async_ctx + resume_offset;
```

Comment:
Given that you show the offsets of Parent and ResumeParent in the comment, is it clarifying to have `resume_offset`.  Why not just `addr_t resume_ptr = async_ctx + ptr_size`.  (also why is `resume_ptr` `uint32_t`, it's adding an addr_t and an offset isn't it?)

Reply:
> also why is resume_ptr uint32_t, it's adding an addr_t and an offset isn't it?

yep you caught a bug, it was after I switched these variables away from `auto` types.

Reply:
That's one of the reasons I will die on the proverbial `auto` hill 👀

Reply:
I'm with you